### ISSUE TITLE
SLiM 3.7.1 release rollout

### ISSUE THREAD

**Issue by bhaller** [MesserLab/SLiM#286]
Hi all.  SLiM 3.7.1 has been frozen and tagged, but nothing has been announced yet; we're mid-release.  I have built the macOS installer and it is ready for uploading; I plan to post it on Monday, and announce the release on slim-discuss then.  In the meantime, I'm trying something new: a GitHub issue to track the progress of the release, since it is getting increasingly complicated to coordinate everything.  So, here are other tasks related to the release:

- [x] Write up the description of the release for the slim-discuss announcement and put it into the GitHub release description.

- [x] Turn the pacman crank for Windows.  I have built the new pacman archive and put it online in the usual spot.  @rdinnager, I believe it is now ready for you to initiate the turn of the crank.

- [x] Turn the crank for Arch.  @grahamgower it'd be great if you could do that when you've got time.

- [x] Turn the crank for the other Linux installers.  @bryce-carson If you could turn the Copr crank this time that would be great; in any case I think you probably need to do something with your Focal Fossa installer too.

- [x] @bryce-carson re: the manual update for the Focal Fossa installer, we can finalize it and roll it into the release without disturbing other aspects of the rollout, thanks.

- [x] Update conda.  Their new auto-update bots made this super easy (yay!) and it appears to have been successful for macOS, Linux, and Windows (https://anaconda.org/conda-forge/slim).  If someone who uses conda would like to verify that it now installs 3.7.1, that would be nice, but I'm not worried, it looks clean.

- [x] @petrelharp I'll tag you here since you may be interested, but as far as I know nothing is needed on the pyslim end for this release.  :->

Please post a reply on this issue when your respective crank turn has begun/completed, so I can track what's going on.  I'll post a first announcement on slim-discuss on Monday when I release for macOS/conda/Linux, and a second time when the Windows installer is ready since it takes much longer.  Thanks to all of you for your help!

**Comment by bhaller** [comment]
The release has now been posted to GitHub, with release notes.  https://github.com/MesserLab/SLiM/releases/tag/v3.7.1

**Comment by bryce-carson** [comment]
### Fedora, Red Hat Enterprise, CentOS, and openSUSE

- [ ] Build & package locally
- [ ] Build & package on Copr

### Debian and Ubuntu

- [ ] Shell Script depends on [http://benhaller.com/slim/SLiM.zip](http://benhaller.com/slim/SLiM.zip); expected update Monday
- [ ] Build & package locally: Ubuntu 21.04 Focal Fossa
- [ ] Build & package on Launchpad: Ubuntu 21.04 Focal Fossa

### Documentation

- [ ] Inform users that the shell script is preferred, but that DEB packages for their version of Ubuntu may be available on Launchpad
- [ ] Return document with tracked changes to @bhaller

**Comment by bhaller** [comment]
> ### Fedora, Red Hat Enterprise, CentOS, and openSUSE
> ### Debian and Ubuntu
> ...

OK, if I understand correctly all of your installers are either usable now, or will automatically be usable as soon as I update http://benhaller.com/slim/SLiM.zip.  Awesome, thanks!

To that end, I have just updated http://benhaller.com/slim/SLiM.zip.  It's a bit odd to update it before I've officially announced the release – maybe we need a better workflow here somehow – but for this release it seems OK since this is such a minor bug-fix release.  People who download that file will now get the 3.7.1 sources, but that's not the end of the world – they probably actually want them, anyhow.  :->  So if you want to test the installer script with the new source package, you can now do so.  Thanks!

> ### Documentation
> > Re: the manual update for the Focal Fossa installer, I think we can just announce it on slim-discuss once it is ready, and add it to the manual for the following release; I really needed to freeze this release today, sorry.
> 
> I missed this, or misunderstood it when I read it earlier. No worries, Ben. I do have the changes nearly ready if you'd like to incorporate them. I'll send the email along shortly.
> 
> * [x]  Inform users that the shell script is preferred, but that DEB packages for their version of Ubuntu may be available on Launchpad
> * [x]  Return document with tracked changes to @bhaller

Got it.  Since the rollout has been going so smoothly, I may be able to incorporate the new doc into the release.  Only the macOS installer includes the manual PDFs, and they're not part of the tagged GitHub release, so I can still change them pretty easily.  I think I'm going to do some minor rewriting of what you added; I'll send you my rewrite by email shortly, we can communicate about it over there.  :->

(I was worried that this update would be bumpy because I'm actually in Mexico right now, and so I needed to remote-login to two different Mac minis back in New York in order to produce the necessary binaries for the macOS installer, because the process for making that installer is such a nightmare – requiring different versions of macOS, Xcode, the macOS SDK, etc., for different builds due to Qt, Apple Silicon, etc., ugh.  But that remote login craziness actually went totally smoothly, so all is well, and now I have a little breathing room.  :->)

**Comment by bryce-carson** [comment]
> OK, if I understand correctly all of your installers are either usable now, or will automatically be usable as soon as I update http://benhaller.com/slim/SLiM.zip. Awesome, thanks!

All of the packages have been updated, yep! The _only_ one of them that depends on the ZIP file on your website is `DebianUbuntuInstaller.sh`, because that was just the most sensible place to always get the _latest_ ZIP without needing to increment the version in a GitHub tag.

> I needed to remote-login to two different Mac minis

Hopefully you had some SSH tunnels setup! Remote desktop and whatnot can be a pain, especially over slow protocols!

**Comment by bhaller** [comment]
Did some retrospective editing above to turn my initial post into a checkbox list; I've never used those before!

**Comment by bryce-carson** [comment]
> Did some retrospective editing above to turn my initial post into a checkbox list; I've never used those before!

They're nice! I just noticed that once you did that the issue header (in the top of the browser view) shows the number of open tasks (assuming all of the checkboxes in the original post are tasks...)! Neato!

**Comment by bhaller** [comment]
> All of the packages have been updated, yep! The _only_ one of them that depends on the ZIP file on your website is `DebianUbuntuInstaller.sh`, because that was just the most sensible place to always get the _latest_ ZIP without needing to increment the version in a GitHub tag.

Awesome!  Maybe for next rollout, we should make a private copy of SLiM.zip on the website that the installer uses but that is not otherwise visible, so that we can update that private copy to a new version without changing the version of the public download?  What do you think of that?

> > I needed to remote-login to two different Mac minis
> 
> Hopefully you had some SSH tunnels setup! Remote desktop and whatnot can be a pain, especially over slow protocols!

Ha, I wish I was that competent with Unix-y stuff.  No, I have no idea how to do that.  I needed to have screen-sharing and graphical remote control of the machines to do the work; maybe everything I did is doable in the CLI somehow, but that would be a *major* project for me.  Happily, my wife is back in NY, so she could start up the machines (which were not even booted), set up zoom on them, accept zoom invitations, share their screens, and accept my remote control request.  With her at the other end, it went very smoothly.  I've never attempted such a thing before, so I was rather worried that it wouldn't work.  I didn't anticipate needing to do this while I was out of town, this was totally unplanned and impromptu.  :-O  I would love to be able to set the machines up to auto-accept zoom invitations and remote control requests from me, so that I could do all of this without needing a human being at the other end; but from the googling I've done so far, that does not appear to be possible.  There's a thing called Apple Remote Desktop that might allow me to do this, but the machines do not have permanent static IP addresses, and so I'm not sure how to configure ARD to be able to find them.  Mysteries abound.  :->

**Comment by bryce-carson** [comment]
> > All of the packages have been updated, yep! The _only_ one of them that depends on the ZIP file on your website is `DebianUbuntuInstaller.sh`, because that was just the most sensible place to always get the _latest_ ZIP without needing to increment the version in a GitHub tag.
> 
> Awesome! Maybe for next rollout, we should make a private copy of SLiM.zip on the website that the installer uses but that is not otherwise visible, so that we can update that private copy to a new version without changing the version of the public download? What do you think of that?

Whatever would be the least disruptive to end users, I imagine. I don't think having an unlisted file just for the installation script is really worth it. If you're not comfortable with someone downloading a more recent version of SLiM a few days before its release without knowing it, it's up to you to disallow that. :) I think in the vanishingly small number of cases where that would happen (and I believe most of them would happen through the installation script as well) is likely a good thing rather than bad. :)

**Comment by bhaller** [comment]
> Whatever would be the least disruptive to end users, I imagine. I don't think having an unlisted file just for the installation script is really worth it. If you're not comfortable with someone downloading a more recent version of SLiM a few days before its release without knowing it, it's up to you to disallow that. :) I think in the vanishingly small number of cases where that would happen (and I believe most of them would happen through the installation script as well) is likely a good thing rather than bad. :)

It's the "without knowing it" part that bothers me – and yes, even if we had a private copy of SLiM.zip for the installer's use, people would still end up installing a new version without knowing it if they used the installer script, so that is not a complete solution.  Hmm.  Well, it comes down to whether we need to test the installer script before going public with a release, or not.  If we don't, then I can switch SLiM.zip over at the same time that I make the public announcement.  If we do, then perhaps we need (a) a separate private copy of SLiM.zip for pre-release testing, _and_ (b) a separate copy of the installer script, otherwise identical, that references that private copy.  Then, pre-release, we can change the private SLiM.zip, run the private installer script, and verify that it works; and then when it's time to go public I would replace the public SLiM.zip.  But it is possible that I am over-thinking this.  :->  The simple fact, which I am uncomfortable with but seems inevitable, is that it is not possible to do a clean "flip of a switch" release process where everything switches over to the new version simultaneously with the release announcement.  Conda is already switched over now, as are the Linux installers; macOS will switch over when I post on Monday; Windows might take another week because of the process needed for it.  That's the way it goes, even though the control freak in me doesn't like it.  Deep breaths.  :->

**Comment by bryce-carson** [comment]
> If we don't, then I can switch SLiM.zip over at the same time that I make the public announcement.

Thankfully, the script is pretty robust being that it only does a few checks with `dpgk`, then builds with `cmake`, and installs the desktop files and binaries to the appropriate places. No testing needed. :)

**Comment by bhaller** [comment]
SLiM 3.7.1 has just been officially released!  Thanks to all of you for your help with this release.  I'll close this issue once the Windows installer is available, just so I don't forget about that.  :->

**Comment by bhaller** [comment]
@rdinnager I take it the pacman crank is still turning?  I forget – is there a way for me to check that myself so I don't have to bug you?  :->

**Comment by rdinnager** [comment]
As far as I know the only way to check is to try and install it (after making sure pacman is up to date) and then check the version number (so you can only do that if you have a windows machine handy). I have just done that and yes, 3.7.1 is the version being installed now. So should be good to close this.

**Comment by bhaller** [comment]
> As far as I know the only way to check is to try and install it (after making sure pacman is up to date) and then check the version number (so you can only do that if you have a windows machine handy). I have just done that and yes, 3.7.1 is the version being installed now. So should be good to close this.

Aha, ok.  Thanks, closing.
